src/components/Article.jsx

```diff
@@ -35,9 +44,54 @@ const {articleId} = useParams();
         }) 
     }, [articleId])
 
-useEffect(() => {
 
-})    
+//voting functions
+const upVote = () => {
+   setUserVote((currVote) => {
+    if (currVote === 1) {
+        setVoteMessage("You can only upvote once!")
+        return 1
+    } else if (currVote === -1) {
+        setVoteMessage("Your vote has been reset to neutral, click again to upvote")
+    } else {
+        setVoteMessage("Thanks for giving your feedback!")
+    }
+
+    changeArticleVote(articleId, 1)
+    .catch(() => {
+        setVoteMessage("Oops, your vote hasn't worked, try again")
+        return currVote
+    })
+
+    return ++currVote;
+   })
+}
+
```

Comment:
Some complicated logic going on here but the UI outcome for limiting votes (up and down) is very satisfying! Nice work.